lib/functions/plot_measures.R

```diff
@@ -29,62 +29,99 @@ plot_measures <- function(
     facet_wrap = FALSE,
     facet_var = NULL,
     colour_var = NULL,
+    shape_var = NULL,
+    save_path = NULL,
+    colour_palette = NULL,
+    y_scale = NULL,
+    scale_measure = NULL,
+    shapes = NULL,
+    date_breaks = "1 month",
     legend_position = "bottom") {
   # Test if all columns expected in output from generate measures exist
-  expected_names <- c("measure", "interval_start", "interval_end", "ratio", "numerator", "denominator")
-  missing_columns <- setdiff(expected_names, colnames(data))
+  # expected_names <- c("measure", "interval_start", "interval_end", "ratio", "numerator", "denominator")
+  # missing_columns <- setdiff(expected_names, colnames(data))
 
-  if (length(missing_columns) > 0) {
-    stop("Data does not have expected column(s): ", paste(missing_columns, collapse = ", "), call. = FALSE)
-  }
+  # if (length(missing_columns) > 0) {
+  #   stop("Data does not have expected column(s): ", paste(missing_columns, collapse = ", "), call. = FALSE)
+  # }
 
   plot_tmp <- ggplot(
     data,
     aes(
       x = {{ select_interval_date }},
       y = {{ select_value }},
       colour = {{ colour_var }},
-      group = {{ colour_var }}
+      group = {{ colour_var }},
+      shape = {{ colour_var }},
+      fill = {{ colour_var }}
     )
   ) +
-    geom_point() +
-    geom_line(alpha = .5) +
+    geom_point(size = 2) +
+    geom_line(alpha = .3) +
     geom_vline(
       xintercept = lubridate::as_date("2024-02-01"),
       linetype = "dotted",
       colour = "orange",
       linewidth = .7
     ) +
     scale_x_date(
-      date_breaks = "1 month",
+      date_breaks = {{ date_breaks }},
       labels = scales::label_date_short()
     ) +
     guides(
-      color = guide_legend(nrow = guide_nrow)
+      color = guide_legend(nrow = guide_nrow),
+      shape = guide_legend(nrow = guide_nrow)
     ) +
     labs(
       title = title,
       x = x_label,
       y = y_label,
       colour = guide_label,
+      shape = NULL,
+      fill = NULL
     ) +
     theme(
       legend.position = legend_position,
-      plot.title = element_text(hjust = 0.5)
+      plot.title = element_text(hjust = 0.5),
+      text = element_text(size = 14)
     )
 
+  # Change colour based on specified colour palette
+  if (!is.null(colour_palette)) {
+    if (length(colour_palette) == 1 && colour_palette == "plasma") {
```

Comment:
I'm not entirely sure what's happening here, but it looks like there are two different ways a user can use the `colour_palette` argument:

1. define the option "plasma" in which case the `scale_colour_viridis_d()` with plasma setting is applied
2. specify a manual colour palette that is used in `scale_colour_manual()`

At the moment this isn't very clear, so think we should at least document this, or consider if there is a clearer way to give the user these two options. Maybe I got this wrong?

Reply:
You're right, I'll document this.